app/code/Magento/Sales/Console/Command/PaymentUpdateEncryptCommand.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Sales\Console\Command;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Command\Command;
+use Magento\Framework\Console\Cli;
+use Magento\Framework\Encryption\Encryptor;
+
+/**
+ * Command for updated encrypted credit card data to the latest cipher
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class PaymentUpdateEncryptCommand extends Command
+{
+    /** Command name */
+    const NAME = 'encryption:update:payment';
+    const LEGACY_PATTERN = '^[[:digit:]]+:[^%s]:.*$';
+
+    /**
+     * @var \Magento\Framework\Encryption\Encryptor
+     */
+    private $encryptor;
+
+    /**
+     * @var \Magento\Sales\Model\ResourceModel\Order\Payment
+     */
+    private $paymentResource;
+
+    /**
+     * @param \Magento\Framework\Encryption\Encryptor $encryptor
+     * @param \Magento\Sales\Model\ResourceModel\Order\Payment $paymentResource
+     */
+    public function __construct(
+        \Magento\Framework\Encryption\Encryptor $encryptor,
+        \Magento\Sales\Model\ResourceModel\Order\Payment $paymentResource
+    ) {
+        $this->encryptor = $encryptor;
+        $this->paymentResource = $paymentResource;
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setName(self::NAME)
+            ->setDescription(
+                'Re-encrypts encrypted credit card data with latest encryption cipher.'
+            );
+        parent::configure();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $connection = $this->paymentResource->getConnection();
+        $table = $this->paymentResource->getTable('sales_order_payment');
+        $select = $connection->select()->from($table, ['entity_id', 'cc_number_enc'])
+            ->where(
+                'cc_number_enc REGEXP ?',
+                sprintf(self::LEGACY_PATTERN, Encryptor::CIPHER_LATEST)
+            )->limit(1000);
+
+        while ($attributeValues = $connection->fetchPairs($select)) {
+            $connection->beginTransaction();
```

Comment:
As the command can be run by somebody on a live store, transactions may cause performance issues/locks.
I propose to remove the transaction, and add another condition to the update, so that we guaranteed do not replace data that a user could enter if it happens between we fetch the data and update it. Something like this:
replace `['entity_id = ?' => (int)$valueId]` with `['entity_id = ?' => (int)$valueId, 'and cc_number_enc = ?' => $value]`